Example/PaymentSheet Example/PaymentSheet Example/ExampleSwiftUIEmbeddedPaymentElement.swift

```diff
@@ -0,0 +1,368 @@
+    
+    // MARK: Step 3) Confirm on the server side
+    private func confirmIntent(
+        paymentMethodID: String,
+        shouldSavePaymentMethod: Bool,
+        isSubscribing: Bool
+    ) async throws -> String {
+        var request = URLRequest(url: confirmIntentUrl)
+        request.httpMethod = "POST"
+        request.setValue("application/json", forHTTPHeaderField: "Content-type")
+        
+        let body: [String: Any?] = [
+            "payment_method_id": paymentMethodID,
+            "currency": "USD",
+            "hot_dog_count": 5,
+            "salad_count": 0,
+            "is_subscribing": isSubscribing,
+            "should_save_payment_method": shouldSavePaymentMethod,
+            "return_url": "payments-example://stripe-redirect",
+            "customer_id": self.response?.customerID,
+        ]
+        request.httpBody = try JSONSerialization.data(withJSONObject: body, options: [])
+        
+        let (data, _) = try await URLSession.shared.data(for: request)
+        let json = try JSONDecoder().decode([String: String].self, from: data)
+        
+        if let clientSecret = json["intentClientSecret"] {
+            return clientSecret
+        } else if let error = json["error"] {
+            throw ExampleError(errorDescription: error)
+        } else {
+            throw ExampleError(errorDescription: "Unknown error from server.")
+        }
+    }
+    
+    // MARK: Step 4) If the user changes subscription, we fetch updated totals from /compute_totals
+    @MainActor
+    func fetchUpdatedTotal(isSubscribing: Bool) async throws {
+        var request = URLRequest(url: computeTotalsUrl)
+        request.httpMethod = "POST"
+        request.setValue("application/json", forHTTPHeaderField: "Content-type")
+        
+        let body: [String: Any] = [
+            "hot_dog_count": 5,
+            "salad_count": 0,
+            "is_subscribing": isSubscribing
+        ]
+        request.httpBody = try JSONSerialization.data(withJSONObject: body, options: [])
+        
+        let (data, _) = try await URLSession.shared.data(for: request)
+        guard
+            let json = try JSONSerialization.jsonObject(with: data, options: []) as? [String: Any],
+            let total = json["total"] as? Int
+        else {
+            throw URLError(.badServerResponse)
+        }
+        
+        self.currentTotal = total
+    }
+    
+    struct ExampleError: LocalizedError {
+        var errorDescription: String?
+    }
+}
+
+// MARK: - MyEmbeddedCheckoutView
+@available(iOS 15.0, *)
+struct MyEmbeddedCheckoutView: View {
+    @StateObject var embeddedViewModel = EmbeddedPaymentElementViewModel()
+    @StateObject var backendViewModel = BackendViewModel()
+    @State var confirmationResult: EmbeddedPaymentElementResult?
+    @State private var isSubscribing: Bool = false
+    
+    @Environment(\.dismiss) private var dismiss
+    
+    // MARK: Body
+    var body: some View {
+        VStack(spacing: 24) {
```

Comment:
ooc why VStack here? Seems like there's at most one thing we're displaying, not a list.

Reply:
Not required! I removed the VStack in favor of a high level ScrollView instead.